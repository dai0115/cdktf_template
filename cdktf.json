{
  "language": "typescript",
  "app": "npx ts-node main.ts",
  "projectId": "8f6cdb9b-f601-410e-ace2-9b4a7e619262",
  "sendCrashReports": "false",
  "terraformProviders": [],
  "terraformModules": [],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}
