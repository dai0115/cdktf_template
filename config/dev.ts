import { ConfigType } from "./types";

export const devConfig: ConfigType = {
  region: "ap-northeast-1",
  stage: "dev",
  prefix: "application-dev",
  network: {
    vpcConfig: {
      cidrBlock: "10.0.0.0/16",
      enableDnsHostnames: true,
      enableDnsSupport: true,
    },
  },
};
