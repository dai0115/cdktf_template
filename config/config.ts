import { ConfigType, stageType } from "./types";
import { devConfig } from "./dev";

// TODO stagingとproductionを作る際に@ts-ignoreを削除する
const configMap: { [key in stageType]: ConfigType } = {
  dev: devConfig,
  // @ts-ignore
  staging: {},
  // @ts-ignore
  prduction: {},
};

export const getConfig = (stage: stageType): ConfigType => {
  return configMap[stage];
};
