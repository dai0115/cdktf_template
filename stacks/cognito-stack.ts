import { CognitoUserPool } from "@cdktf/provider-aws/lib/cognito-user-pool";
import { AwsProvider } from "@cdktf/provider-aws/lib/provider";
import { TerraformStack } from "cdktf";
import { Construct } from "constructs";
import { ConfigType } from "../config/types";

type CognitoProps = {
  config: ConfigType;
};

export class CognitoStack extends TerraformStack {
  readonly cognitoId: String;

  constructor(scope: Construct, id: string, props: CognitoProps) {
    super(scope, id);
    const { region, prefix } = props.config;
    const { cognito } = props.config;

    new AwsProvider(this, `${prefix}-AWS`, {
      region: region,
    });

    const cog = new CognitoUserPool(this, `${prefix}-Cognito`, {
      ...cognito,
    });
    this.cognitoId = cog.id;
  }
}
