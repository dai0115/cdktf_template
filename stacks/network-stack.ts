import { AwsProvider } from "@cdktf/provider-aws/lib/provider";
import { TerraformStack } from "cdktf";
import { Construct } from "constructs";

import { Vpc } from "@cdktf/provider-aws/lib/vpc";
import { Subnet } from "@cdktf/provider-aws/lib/subnet";
import { SecurityGroup } from "@cdktf/provider-aws/lib/security-group";
import { SecurityGroupRule } from "@cdktf/provider-aws/lib/security-group-rule";
import { InternetGateway } from "@cdktf/provider-aws/lib/internet-gateway";
import { RouteTable } from "@cdktf/provider-aws/lib/route-table";
import { RouteTableAssociation } from "@cdktf/provider-aws/lib/route-table-association";
import { VpcEndpoint } from "@cdktf/provider-aws/lib/vpc-endpoint";

import { ConfigType } from "../config/types";

export class NetworkStack extends TerraformStack {
  readonly ecsSubnetIds: Array<string>;
  readonly dbSubnetIds: Array<string>;
  readonly albSubnetIds: Array<string>;
  readonly endpointSubnetId: string;
  readonly albSG: SecurityGroup;
  readonly ecsSG: SecurityGroup;
  readonly dbSG: SecurityGroup;

  constructor(scope: Construct, id: string, props: ConfigType) {
    super(scope, id);

    const { region, prefix } = props;
    const { vpcConfig } = props.network;

    new AwsProvider(this, `${prefix}-AWS`, {
      region: region,
    });

    // VPCリソースの作成
    const vpc = new Vpc(this, `${prefix}-Vpc`, {
      ...vpcConfig,
      tags: {
        Name: `${prefix}-Vpc`,
      },
    });

    // サブネットの作成
    const ecsSubnet1 = this.createSubnet(
      `${prefix}-Ecs1`,
      vpc.id,
      "10.0.0.0/24",
      "ap-northeast-1a"
    );
    const ecsSubnet2 = this.createSubnet(
      `${prefix}-Ecs2`,
      vpc.id,
      "10.0.1.0/24",
      "ap-northeast-1c"
    );
    this.ecsSubnetIds = [ecsSubnet1, ecsSubnet2];
    const dbSubnet1 = this.createSubnet(
      `${prefix}-Db1`,
      vpc.id,
      "10.0.2.0/24",
      "ap-northeast-1a"
    );
    const dbSubnet2 = this.createSubnet(
      `${prefix}-Db2`,
      vpc.id,
      "10.0.3.0/24",
      "ap-northeast-1c"
    );
    this.dbSubnetIds = [dbSubnet1, dbSubnet2];
    const albSubnet1 = this.createSubnet(
      `${prefix}-Alb1`,
      vpc.id,
      "10.0.4.0/24",
      "ap-northeast-1a"
    );
    const albSubnet2 = this.createSubnet(
      `${prefix}-Alb2`,
      vpc.id,
      "10.0.5.0/24",
      "ap-northeast-1c"
    );
    this.albSubnetIds = [albSubnet1, albSubnet2];
    this.endpointSubnetId = this.createSubnet(
      `${prefix}-Endpoint`,
      vpc.id,
      "10.0.6.0/24",
      "ap-northeast-1a"
    );
    const bastionSubnetId = this.createSubnet(
      `${prefix}-Bastion`,
      vpc.id,
      "10.0.7.0/24",
      "ap-northeast-1a"
    );

    // ルートテーブルの作成
    const ecsRouteTable = this.createRouteTable(`${prefix}-EcsRoute`, vpc);
    const dbRouteTable = this.createRouteTable(`${prefix}-DbRoute`, vpc);
    const albRouteTable = this.createRouteTable(`${prefix}-AlbRoute`, vpc);
    const bastionRouteTable = this.createRouteTable(
      `${prefix}-BastionRoute`,
      vpc
    );
    const endpointRouteTable = this.createRouteTable(
      `${prefix}-EndpointRoute`,
      vpc
    );

    // サブネットとルートテーブルの関連付け
    this.attachRouteTabletoSubnet("ecs1", ecsSubnet1, ecsRouteTable.id);
    this.attachRouteTabletoSubnet("ecs2", ecsSubnet2, ecsRouteTable.id);
    this.attachRouteTabletoSubnet("db1", dbSubnet1, dbRouteTable.id);
    this.attachRouteTabletoSubnet("db2", dbSubnet2, dbRouteTable.id);
    this.attachRouteTabletoSubnet("alb1", albSubnet1, albRouteTable.id);
    this.attachRouteTabletoSubnet("alb2", albSubnet2, albRouteTable.id);
    this.attachRouteTabletoSubnet(
      "bastion",
      bastionSubnetId,
      bastionRouteTable.id
    );
    this.attachRouteTabletoSubnet(
      "endpoint",
      this.endpointSubnetId,
      endpointRouteTable.id
    );

    // パブリックサブネットへのリソースの関連付け
    const iGW = this.createIGW(prefix, vpc.id);
    new RouteTableAssociation(this, "ecs-IGW", {
      gatewayId: iGW.id,
      routeTableId: albRouteTable.id,
    });

    // セキュリティグループの作成
    this.albSG = this.createSecurityGroup("alb", vpc.id);
    new SecurityGroupRule(this, "albSG-ingress-rule", {
      cidrBlocks: ["0.0.0.0/0"],
      type: "ingress",
      fromPort: 80, //TODO replace here to 443
      toPort: 80, //TODO replace here to 443
      protocol: "tcp",
      securityGroupId: this.albSG.id,
      description: "IGW to Load balancer",
    });
    this.attachAllTrafficRules("alb", this.albSG.id);

    this.ecsSG = this.createSecurityGroup("ecs", vpc.id);
    new SecurityGroupRule(this, "ecsSG-ingress-rule", {
      sourceSecurityGroupId: this.albSG.id,
      type: "ingress",
      fromPort: 80,
      toPort: 80,
      protocol: "tcp",
      securityGroupId: this.ecsSG.id,
      description: "Load balancer to target",
    });
    this.attachAllTrafficRules("ecs", this.ecsSG.id);

    this.dbSG = this.createSecurityGroup("db", vpc.id);
    new SecurityGroupRule(this, "dbSG-ingress-rule2", {
      sourceSecurityGroupId: this.ecsSG.id,
      type: "ingress",
      fromPort: 3306,
      toPort: 3306,
      protocol: "tcp",
      securityGroupId: this.dbSG.id,
    });
    this.attachAllTrafficRules("db", this.dbSG.id);

    const endpointSG = this.createSecurityGroup("endpoint", vpc.id);
    new SecurityGroupRule(this, "endpointSG-ingress-rule", {
      cidrBlocks: ["10.0.0.0/16"],
      type: "ingress",
      fromPort: 443,
      toPort: 443,
      protocol: "tcp",
      securityGroupId: endpointSG.id,
    });
    this.attachAllTrafficRules("endpoint", endpointSG.id);

    const bastionSG = this.createSecurityGroup("bastion", vpc.id);
    this.attachAllTrafficRules("bastion", bastionSG.id);

    // VPCエンドポイントの作成
    this.createGatewayEndpoint("ecs", vpc.id, ecsRouteTable.id, "s3");
    this.createGatewayEndpoint("bastion", vpc.id, bastionRouteTable.id, "s3");

    const serviceNames: string[] = [
      "ssm",
      "ssmmessages",
      "ec2messages",
      "ecr.api",
      "ecr.dkr",
      "logs",
      "secretsmanager",
      "ecs-agent",
      "ecs-telemetry",
      "ecs",
    ];
    for (const service of serviceNames) {
      this.createInterfaceEndpoint(
        vpc.id,
        endpointSG.id,
        this.endpointSubnetId,
        service
      );
    }
  }

  /**
   * @param name - 作成するサブネットの名前
   * @param vpcId - サブネットを作成するvpcのId
   * @param cideBlock - 作成するサブネットのcideBlock
   * @param availabilityZone - サブネットを作成するavailabilityZone
   * @returns 作成されたSubnetを返却
   */
  private createSubnet(
    name: string,
    vpcId: string,
    cideBlock: string,
    availabilityZone: string
  ): string {
    const subnet = new Subnet(this, name, {
      vpcId: vpcId,
      cidrBlock: cideBlock,
      availabilityZone: availabilityZone,
      tags: {
        Name: name,
      },
    });
    return subnet.id;
  }

  /**
   * @param name - 作成するルートテーブルの名前
   * @param vpcId - ルートテーブルを作成するvpcのId
   * @returns 作成されたRouteTableを返却
   */
  private createRouteTable(name: string, vpc: Vpc): RouteTable {
    return new RouteTable(this, name, {
      vpcId: vpc.id,
      route: [],
      tags: {
        Name: name,
      },
    });
  }

  /**
   * @param name - 命名時に利用するprefix
   * @param subnetId - ルートテーブルと関連付けを行うサブネットのId
   * @param subnetId - サブネットと関連付けを行うルートテーブルのId
   * 受け取ったサブネットとルートテーブルを紐付ける
   */
  private attachRouteTabletoSubnet(
    name: string,
    subnetId: string,
    routeTableId: string
  ) {
    new RouteTableAssociation(this, `${name}-Association`, {
      subnetId: subnetId,
      routeTableId: routeTableId,
    });
  }

  /**
   * @param name - リソースの命名に利用するprefix
   * @param vpcId - IGWを作成するvpcのId
   * @returns 作成されたInternet Gatewayを返却
   */
  private createIGW(name: string, vpcId: string): InternetGateway {
    return new InternetGateway(this, `${name}-IGW`, {
      vpcId: vpcId,
      tags: {
        Name: `${name}-IGW`,
      },
    });
  }

  /**
   * @param name - セキュリティグループの名前
   * @param vpcId - セキュリティグループを作成するvpcのId
   */
  private createSecurityGroup(name: string, vpcId: string): SecurityGroup {
    return new SecurityGroup(this, `${name}SG`, {
      name: `${name}SG`,
      vpcId: vpcId,
      tags: {
        Name: `${name}SG`,
      },
    });
  }
  /**
   * @param name - セキュリティグループルールの名前
   * @param securityGroupId - ルールを紐付けるセキュリティグループのId
   *
   * 全ての通信を許可する場合のセキュリティグループルールを作成
   */
  private attachAllTrafficRules(name: string, securityGroupId: string) {
    new SecurityGroupRule(this, `${name}SG-egress-rule`, {
      cidrBlocks: ["0.0.0.0/0"],
      type: "egress",
      fromPort: 0,
      toPort: 0,
      protocol: "-1",
      securityGroupId: securityGroupId,
    });
  }
  /**
   * @param vpcId - Vpcエンドポイントを作成するVpcのId
   * @param routeTableIds - VpcエンドポイントGatewayタイプに紐付けるルートテーブルのId
   * @param serviceName - 作成するサービスの名前を指定。"s3"や"ssm"などの形式で"com.amazon〜"は不要
   * @returns 作成したVPCエンドポイントを返却
   *
   * ゲートウェイ型のエンドポイントを作成
   */
  private createGatewayEndpoint(
    name: string,
    vpcId: string,
    routeTableId: string,
    serviceName: string
  ): VpcEndpoint {
    return new VpcEndpoint(this, `${name}-vpc-endpoint-${serviceName}`, {
      vpcId: vpcId,
      serviceName: `com.amazonaws.ap-northeast-1.${serviceName}`,
      vpcEndpointType: "Gateway",
      routeTableIds: [routeTableId],
      tags: {
        Name: `vpc-endpoint-${serviceName}-Gateway`,
      },
    });
  }
  /**
   * @param vpcId - vpcエンドポイントを作成するVpcのId
   * @param securityGroupIds - Vpcエンドポイントに作成するセキュリティグループのId
   * @param subnetIds - Vpcエンドポイントを作成するサブネットのId
   * @param serviceName - 作成するサービスの名前を指定。"s3"や"ssm"などの形式で"com.amazon〜"は不要
   * @returns 作成したVPCエンドポイントを返却
   *
   * インターフェース型のエンドポイントを作成
   */
  private createInterfaceEndpoint(
    vpcId: string,
    securityGroupId: string,
    subnetId: string,
    serviceName: string
  ): VpcEndpoint {
    return new VpcEndpoint(this, `vpc-endpoint-${serviceName}`, {
      vpcId: vpcId,
      serviceName: `com.amazonaws.ap-northeast-1.${serviceName}`,
      vpcEndpointType: "Interface",
      subnetIds: [subnetId],
      securityGroupIds: [securityGroupId],
      privateDnsEnabled: true,
      tags: {
        Name: `vpc-endpoint-${serviceName}`,
      },
    });
  }
}
