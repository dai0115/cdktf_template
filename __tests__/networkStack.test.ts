import { Testing } from "cdktf";
import { NetworkStack } from "../stacks/network-stack";
import { devConfig } from "../config/dev";

describe("networkStack", () => {
  it("snapshot test for vpcStack", () => {
    const app = Testing.app();
    const vpc = new NetworkStack(app, "networkStack", devConfig);
    expect(Testing.synth(vpc)).toMatchSnapshot();
  });
});
