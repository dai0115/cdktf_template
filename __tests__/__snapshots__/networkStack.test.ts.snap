// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`networkStack snapshot test for vpcStack 1`] = `
"{
  "provider": {
    "aws": [
      {
        "region": "ap-northeast-1"
      }
    ]
  },
  "resource": {
    "aws_internet_gateway": {
      "application-dev-IGW": {
        "tags": {
          "Name": "application-dev-IGW"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      }
    },
    "aws_route_table": {
      "application-dev-AlbRoute": {
        "route": [
        ],
        "tags": {
          "Name": "application-dev-AlbRoute"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-BastionRoute": {
        "route": [
        ],
        "tags": {
          "Name": "application-dev-BastionRoute"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-DbRoute": {
        "route": [
        ],
        "tags": {
          "Name": "application-dev-DbRoute"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-EcsRoute": {
        "route": [
        ],
        "tags": {
          "Name": "application-dev-EcsRoute"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-EndpointRoute": {
        "route": [
        ],
        "tags": {
          "Name": "application-dev-EndpointRoute"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      }
    },
    "aws_route_table_association": {
      "alb1-Association": {
        "route_table_id": "\${aws_route_table.application-dev-AlbRoute.id}",
        "subnet_id": "\${aws_subnet.application-dev-Alb1.id}"
      },
      "alb2-Association": {
        "route_table_id": "\${aws_route_table.application-dev-AlbRoute.id}",
        "subnet_id": "\${aws_subnet.application-dev-Alb2.id}"
      },
      "bastion-Association": {
        "route_table_id": "\${aws_route_table.application-dev-BastionRoute.id}",
        "subnet_id": "\${aws_subnet.application-dev-Bastion.id}"
      },
      "db1-Association": {
        "route_table_id": "\${aws_route_table.application-dev-DbRoute.id}",
        "subnet_id": "\${aws_subnet.application-dev-Db1.id}"
      },
      "db2-Association": {
        "route_table_id": "\${aws_route_table.application-dev-DbRoute.id}",
        "subnet_id": "\${aws_subnet.application-dev-Db2.id}"
      },
      "ecs-IGW": {
        "gateway_id": "\${aws_internet_gateway.application-dev-IGW.id}",
        "route_table_id": "\${aws_route_table.application-dev-AlbRoute.id}"
      },
      "ecs1-Association": {
        "route_table_id": "\${aws_route_table.application-dev-EcsRoute.id}",
        "subnet_id": "\${aws_subnet.application-dev-Ecs1.id}"
      },
      "ecs2-Association": {
        "route_table_id": "\${aws_route_table.application-dev-EcsRoute.id}",
        "subnet_id": "\${aws_subnet.application-dev-Ecs2.id}"
      },
      "endpoint-Association": {
        "route_table_id": "\${aws_route_table.application-dev-EndpointRoute.id}",
        "subnet_id": "\${aws_subnet.application-dev-Endpoint.id}"
      }
    },
    "aws_security_group": {
      "albSG": {
        "name": "albSG",
        "tags": {
          "Name": "albSG"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "bastionSG": {
        "name": "bastionSG",
        "tags": {
          "Name": "bastionSG"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "dbSG": {
        "name": "dbSG",
        "tags": {
          "Name": "dbSG"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "ecsSG": {
        "name": "ecsSG",
        "tags": {
          "Name": "ecsSG"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "endpointSG": {
        "name": "endpointSG",
        "tags": {
          "Name": "endpointSG"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      }
    },
    "aws_security_group_rule": {
      "albSG-egress-rule": {
        "cidr_blocks": [
          "0.0.0.0/0"
        ],
        "from_port": 0,
        "protocol": "-1",
        "security_group_id": "\${aws_security_group.albSG.id}",
        "to_port": 0,
        "type": "egress"
      },
      "albSG-ingress-rule": {
        "cidr_blocks": [
          "0.0.0.0/0"
        ],
        "description": "IGW to Load balancer",
        "from_port": 80,
        "protocol": "tcp",
        "security_group_id": "\${aws_security_group.albSG.id}",
        "to_port": 80,
        "type": "ingress"
      },
      "bastionSG-egress-rule": {
        "cidr_blocks": [
          "0.0.0.0/0"
        ],
        "from_port": 0,
        "protocol": "-1",
        "security_group_id": "\${aws_security_group.bastionSG.id}",
        "to_port": 0,
        "type": "egress"
      },
      "dbSG-egress-rule": {
        "cidr_blocks": [
          "0.0.0.0/0"
        ],
        "from_port": 0,
        "protocol": "-1",
        "security_group_id": "\${aws_security_group.dbSG.id}",
        "to_port": 0,
        "type": "egress"
      },
      "dbSG-ingress-rule2": {
        "from_port": 3306,
        "protocol": "tcp",
        "security_group_id": "\${aws_security_group.dbSG.id}",
        "source_security_group_id": "\${aws_security_group.ecsSG.id}",
        "to_port": 3306,
        "type": "ingress"
      },
      "ecsSG-egress-rule": {
        "cidr_blocks": [
          "0.0.0.0/0"
        ],
        "from_port": 0,
        "protocol": "-1",
        "security_group_id": "\${aws_security_group.ecsSG.id}",
        "to_port": 0,
        "type": "egress"
      },
      "ecsSG-ingress-rule": {
        "description": "Load balancer to target",
        "from_port": 80,
        "protocol": "tcp",
        "security_group_id": "\${aws_security_group.ecsSG.id}",
        "source_security_group_id": "\${aws_security_group.albSG.id}",
        "to_port": 80,
        "type": "ingress"
      },
      "endpointSG-egress-rule": {
        "cidr_blocks": [
          "0.0.0.0/0"
        ],
        "from_port": 0,
        "protocol": "-1",
        "security_group_id": "\${aws_security_group.endpointSG.id}",
        "to_port": 0,
        "type": "egress"
      },
      "endpointSG-ingress-rule": {
        "cidr_blocks": [
          "10.0.0.0/16"
        ],
        "from_port": 443,
        "protocol": "tcp",
        "security_group_id": "\${aws_security_group.endpointSG.id}",
        "to_port": 443,
        "type": "ingress"
      }
    },
    "aws_subnet": {
      "application-dev-Alb1": {
        "availability_zone": "ap-northeast-1a",
        "cidr_block": "10.0.4.0/24",
        "tags": {
          "Name": "application-dev-Alb1"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-Alb2": {
        "availability_zone": "ap-northeast-1c",
        "cidr_block": "10.0.5.0/24",
        "tags": {
          "Name": "application-dev-Alb2"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-Bastion": {
        "availability_zone": "ap-northeast-1a",
        "cidr_block": "10.0.7.0/24",
        "tags": {
          "Name": "application-dev-Bastion"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-Db1": {
        "availability_zone": "ap-northeast-1a",
        "cidr_block": "10.0.2.0/24",
        "tags": {
          "Name": "application-dev-Db1"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-Db2": {
        "availability_zone": "ap-northeast-1c",
        "cidr_block": "10.0.3.0/24",
        "tags": {
          "Name": "application-dev-Db2"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-Ecs1": {
        "availability_zone": "ap-northeast-1a",
        "cidr_block": "10.0.0.0/24",
        "tags": {
          "Name": "application-dev-Ecs1"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-Ecs2": {
        "availability_zone": "ap-northeast-1c",
        "cidr_block": "10.0.1.0/24",
        "tags": {
          "Name": "application-dev-Ecs2"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "application-dev-Endpoint": {
        "availability_zone": "ap-northeast-1a",
        "cidr_block": "10.0.6.0/24",
        "tags": {
          "Name": "application-dev-Endpoint"
        },
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      }
    },
    "aws_vpc": {
      "application-dev-Vpc": {
        "cidr_block": "10.0.0.0/16",
        "enable_dns_hostnames": true,
        "enable_dns_support": true,
        "tags": {
          "Name": "application-dev-Vpc"
        }
      }
    },
    "aws_vpc_endpoint": {
      "bastion-vpc-endpoint-s3": {
        "route_table_ids": [
          "\${aws_route_table.application-dev-BastionRoute.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.s3",
        "tags": {
          "Name": "vpc-endpoint-s3-Gateway"
        },
        "vpc_endpoint_type": "Gateway",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "ecs-vpc-endpoint-s3": {
        "route_table_ids": [
          "\${aws_route_table.application-dev-EcsRoute.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.s3",
        "tags": {
          "Name": "vpc-endpoint-s3-Gateway"
        },
        "vpc_endpoint_type": "Gateway",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "vpc-endpoint-ec2messages": {
        "private_dns_enabled": true,
        "security_group_ids": [
          "\${aws_security_group.endpointSG.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.ec2messages",
        "subnet_ids": [
          "\${aws_subnet.application-dev-Endpoint.id}"
        ],
        "tags": {
          "Name": "vpc-endpoint-ec2messages"
        },
        "vpc_endpoint_type": "Interface",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "vpc-endpoint-ecrapi": {
        "private_dns_enabled": true,
        "security_group_ids": [
          "\${aws_security_group.endpointSG.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.ecr.api",
        "subnet_ids": [
          "\${aws_subnet.application-dev-Endpoint.id}"
        ],
        "tags": {
          "Name": "vpc-endpoint-ecr.api"
        },
        "vpc_endpoint_type": "Interface",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "vpc-endpoint-ecrdkr": {
        "private_dns_enabled": true,
        "security_group_ids": [
          "\${aws_security_group.endpointSG.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.ecr.dkr",
        "subnet_ids": [
          "\${aws_subnet.application-dev-Endpoint.id}"
        ],
        "tags": {
          "Name": "vpc-endpoint-ecr.dkr"
        },
        "vpc_endpoint_type": "Interface",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "vpc-endpoint-ecs": {
        "private_dns_enabled": true,
        "security_group_ids": [
          "\${aws_security_group.endpointSG.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.ecs",
        "subnet_ids": [
          "\${aws_subnet.application-dev-Endpoint.id}"
        ],
        "tags": {
          "Name": "vpc-endpoint-ecs"
        },
        "vpc_endpoint_type": "Interface",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "vpc-endpoint-ecs-agent": {
        "private_dns_enabled": true,
        "security_group_ids": [
          "\${aws_security_group.endpointSG.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.ecs-agent",
        "subnet_ids": [
          "\${aws_subnet.application-dev-Endpoint.id}"
        ],
        "tags": {
          "Name": "vpc-endpoint-ecs-agent"
        },
        "vpc_endpoint_type": "Interface",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "vpc-endpoint-ecs-telemetry": {
        "private_dns_enabled": true,
        "security_group_ids": [
          "\${aws_security_group.endpointSG.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.ecs-telemetry",
        "subnet_ids": [
          "\${aws_subnet.application-dev-Endpoint.id}"
        ],
        "tags": {
          "Name": "vpc-endpoint-ecs-telemetry"
        },
        "vpc_endpoint_type": "Interface",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "vpc-endpoint-logs": {
        "private_dns_enabled": true,
        "security_group_ids": [
          "\${aws_security_group.endpointSG.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.logs",
        "subnet_ids": [
          "\${aws_subnet.application-dev-Endpoint.id}"
        ],
        "tags": {
          "Name": "vpc-endpoint-logs"
        },
        "vpc_endpoint_type": "Interface",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "vpc-endpoint-secretsmanager": {
        "private_dns_enabled": true,
        "security_group_ids": [
          "\${aws_security_group.endpointSG.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.secretsmanager",
        "subnet_ids": [
          "\${aws_subnet.application-dev-Endpoint.id}"
        ],
        "tags": {
          "Name": "vpc-endpoint-secretsmanager"
        },
        "vpc_endpoint_type": "Interface",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "vpc-endpoint-ssm": {
        "private_dns_enabled": true,
        "security_group_ids": [
          "\${aws_security_group.endpointSG.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.ssm",
        "subnet_ids": [
          "\${aws_subnet.application-dev-Endpoint.id}"
        ],
        "tags": {
          "Name": "vpc-endpoint-ssm"
        },
        "vpc_endpoint_type": "Interface",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      },
      "vpc-endpoint-ssmmessages": {
        "private_dns_enabled": true,
        "security_group_ids": [
          "\${aws_security_group.endpointSG.id}"
        ],
        "service_name": "com.amazonaws.ap-northeast-1.ssmmessages",
        "subnet_ids": [
          "\${aws_subnet.application-dev-Endpoint.id}"
        ],
        "tags": {
          "Name": "vpc-endpoint-ssmmessages"
        },
        "vpc_endpoint_type": "Interface",
        "vpc_id": "\${aws_vpc.application-dev-Vpc.id}"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "aws": {
        "source": "aws",
        "version": "4.60.0"
      }
    }
  }
}"
`;
